docker/generic/start_proxy.py

```diff
@@ -394,7 +398,34 @@ def make_argparser():
         | /hello///        | Rejected              | /hello             |
         -----------------------------------------------------------------
         
-        By default, ESPv2 will not merge slashes.''')
+        By default, ESPv2 will merge slashes.
+        Disable the feature only if your traffic is affected by the behavior.
+        
+        For more details, see:
+        https://cloud.google.com/api-gateway/docs/path-templating''')
+
+    parser.add_argument('--escape_slashes_in_path', action='store_true',
+        help='''Enables escaping of the following characters in the `path`
+        HTTP header:
+        - %%2F or %%2f -> /
+        - %%5C or %%5c -> \
+        
+        When enabled, the behavior depends on the protocol:
+        - For OpenAPI backends, request paths with escaped slashes will be
+          automatically unescaped via a redirect.
+        - For gRPC backends, request paths with escaped slashes will be
+          rejected.
```

Comment:
not clear to me. 

if true,   %2f will be still %2f when received by the backend, is that correct?  if so,  should this be called `keep_escaped_slash_in_path`

Reply:
No, if this flag is set, the %2f will be converted to a `/` character. I tried to clarify.

Reply:
I see.  then how about `--unescape_escaped_slashes_in_path` with help as
If there are any escaped slashes in path, e.g. %%2f,  they will be converted to slashes when doing routing.

Reply:
How about `--disallow_escaped_slashes_in_path`? Because gRPC does not support redirects, this flag will rejects gRPC requests that have escaped characters. Only non-gRPC traffic will be automatically unescaped.

Reply:
Cool.  I like it.  Thanks